### PR TITLE
docs(docs-infra): move JSDoc params

### LINKED ISSUE #56349: Angular.dev is missing a lot of information
### Describe the problem that you experienced

Compare:
- https://angular.dev/api/common/CurrencyPipe
- https://v17.angular.io/api/common/CurrencyPipe

The angular.io page provides a detailed explanation for each parameter as well as a code sample.
The angular.dev page is missing parameter explanations and also missing the code sample. 

### Enter the URL of the topic with the problem

https://angular.dev/api/common/CurrencyPipe

### Describe what you want to experience that would fix the problem

Angular.dev should have been ready for production (QA) before it was made the default.

### Add a screenshot if that helps illustrate the problem

https://github.com/angular/angular/assets/5672108/7193f41c-fb2c-4e03-aef3-336b28d00b87

image

### Provide any additional information here in as much as detail as you can

Related: https://github.com/angular/angular/issues/56321

### PR DESCRIPTION
**Pull request by vladboisa**
Move JSDoc params upper, for correct view.
Like in [ADEV docs for SlicePipe](https://angular.dev/api/common/SlicePipe?tab=api) in [source file](https://github.com/angular/angular/blob/main/packages/common/src/pipes/slice_pipe.ts#L48-L95) the params for Pipe going before code
Also all examples for Usage Note now located in `adev/src/content/api-examples/common/pipes/ts` but not in `common/pipes/ts/currency_pipe.ts`

Fixes #56349

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #56349 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/common/src/pipes/number_pipe.ts [hunk 1]
```diff
@@ -198,7 +198,7 @@ export class PercentPipe implements PipeTransform {
  * into text strings, according to various format specifications,
  * where the caller's default locale is `en-US`.
  *
- * <code-example path="common/pipes/ts/currency_pipe.ts" region='CurrencyPipe'></code-example>
+ * <code-example path="api-examples/common/pipes/ts/currency_pipe.ts" region='CurrencyPipe'></code-example>
```
Comment (JeanMeche) [inline packages/common/src/pipes/number_pipe.ts:201]: The fact that this is broken is not a path issue, but a lack of support from our toolchain.

File: packages/common/src/pipes/number_pipe.ts [hunk 2]
```diff
@@ -266,6 +240,32 @@ export class CurrencyPipe implements PipeTransform {
    * When not supplied, uses the value of `LOCALE_ID`, which is `en-US` by default.
    * See [Setting your app locale](guide/i18n/locale-id).
    */
+  constructor(
```
Comment (JeanMeche) [inline packages/common/src/pipes/number_pipe.ts:243]: I feel like the comment block should above the 1st `transform` signature, not the constructor.

**Review by JeanMeche — CHANGES_REQUESTED** [review]
You should probably run the doc site locally with `yarn docs` to test the changes you are making to the docs.

From what I can see, there are no visual changes with this commit.

**Comment by vladboisa** [comment]
> You should probably run the doc site locally with `yarn docs` to test the changes you are making to the docs.
> 
> From what I can see, there are no visual changes with this commit.

Yea, but unfortunately, can't serve locally : #57117

File: packages/common/src/pipes/number_pipe.ts (hunk 1 above)
Comment (vladboisa) [reply packages/common/src/pipes/number_pipe.ts:201]: Okay, got you

File: packages/common/src/pipes/number_pipe.ts (hunk 2 above)
Comment (vladboisa) [reply packages/common/src/pipes/number_pipe.ts]: Yea, it's worked correctly
Thank you

**Pull request closed without merging**